src/tests/ftest/util/dmg_utils.py

```diff
@@ -74,7 +75,61 @@ def storage_scan(self, verbose=False):
         self.set_sub_command("storage")
         self.sub_command_class.set_sub_command("scan")
         self.sub_command_class.sub_command_class.verbose.value = verbose
-        return self._get_result()
+        self.result = self._get_result()
+        self.log.info("--- self.result.stdout ---")
+        self.log.info(self.result.stdout)
+
+        if verbose:
+            vals = re.findall(
```

Comment:
Thanks for suggestion. I implemented as you described. For the hostname, we can't directly use NodeSet; we'll have to create a class and implement some methods like __hash__ to make the hashing work with NodeSet. I used string because it would be easier to access. Also, I think the caller should consider using NodeSet to obtain the values they want.